src/robot/libraries/Telnet.py

```diff
@@ -528,7 +528,9 @@ def __init__(self, host=None, port=23, timeout=3.0, newline='CRLF',
         self._terminal_type = self._encode(terminal_type) if terminal_type else None
         self.set_option_negotiation_callback(self._negotiate_options)
         self._set_telnetlib_log_level(telnetlib_log_level)
-        self._opt_responses = list()
+        self._opt_responses = set()
```

Comment:
convert to set(), so we can use the discard() method instead of `try: l.remove(); except ValueError: pass`